gui/src/app/CompileContext/CompileContextProvider.tsx

```diff
@@ -0,0 +1,102 @@
+import { ProjectContext } from "@SpCore/ProjectContextProvider";
+import compileStanProgram from "@SpCompileContext/compileStanProgram";
+import {
+  FunctionComponent,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { CompileContext, CompileStatus } from "./CompileContext";
+
+type CompileContextProviderProps = {
+  // none
+};
+
+const initialStanWasmServerUrl =
+  localStorage.getItem("stanWasmServerUrl") ||
+  "https://trom-stan-wasm-server.magland.org";
+
+export const CompileContextProvider: FunctionComponent<
+  PropsWithChildren<CompileContextProviderProps>
+> = ({ children }) => {
+  const { data } = useContext(ProjectContext);
+  const [compileStatus, setCompileStatus] = useState<CompileStatus>("");
+  const [
+    theStanFileContentThasHasBeenCompiled,
+    setTheStanFileContentThasHasBeenCompiled,
+  ] = useState<string>("");
+  const [compileMessage, setCompileMessage] = useState<string>("");
+  const [compiledMainJsUrl, setCompiledMainJsUrl] = useState<
+    string | undefined
+  >(undefined);
+  const [validSyntax, setValidSyntax] = useState<boolean>(false);
+
+  useEffect(() => {
+    // if the compiled content is not the same as the current content,
+    // then the state should not be compiled or failed
+    if (data.stanFileContent !== theStanFileContentThasHasBeenCompiled) {
+      if (compileStatus === "compiled" || compileStatus === "failed") {
+        setCompileStatus("");
+        setCompiledMainJsUrl("");
+      }
+    }
+  }, [
+    data.stanFileContent,
+    theStanFileContentThasHasBeenCompiled,
+    compileStatus,
+    setCompiledMainJsUrl,
+  ]);
+
+  const [stanWasmServerUrl, setStanWasmServerUrl] = useState<string>(
+    initialStanWasmServerUrl,
+  );
+  useEffect(() => {
+    // persist to local storage
+    localStorage.setItem("stanWasmServerUrl", stanWasmServerUrl);
+  }, [stanWasmServerUrl]);
```

Comment:
Not sure I understand why we need to persist this to local storage vs. it being part of the context?

Reply:
When user refreshes the page, or closes the tab and comes back later, we want this to remember the choice.

Reply:
(this is also the current behavior, the code just moved in this PR)